mcproto/types/nbt.py

```diff
@@ -0,0 +1,1315 @@
+            ```
+        """
+        if hasattr(data, "to_nbt"):  # For objects that can be converted to NBT
+            return data.to_nbt(name=name)  # type: ignore
+
+        if isinstance(data, int):
+            if -(1 << 7) <= data < 1 << 7:
+                return ByteNBT(data, name=name)
+            if -(1 << 15) <= data < 1 << 15:
+                return ShortNBT(data, name=name)
+            if -(1 << 31) <= data < 1 << 31:
+                return IntNBT(data, name=name)
+            if -(1 << 63) <= data < 1 << 63:
+                return LongNBT(data, name=name)
+            raise ValueError(f"Integer {data} is out of range.")
+        if isinstance(data, float):
+            return FloatNBT(data, name=name)
+        if isinstance(data, str):
+            return StringNBT(data, name=name)
+        if isinstance(data, (bytearray, bytes)):
+            if isinstance(data, bytearray):
+                data = bytes(data)
+            return ByteArrayNBT(data, name=name)
+        if isinstance(data, list):
+            if not data:
+                # Type END is used to mark an empty list
+                return ListNBT([], name=name)
+            first_type = type(data[0])
+            if any(type(item) != first_type for item in data):
+                raise TypeError("All items in a list must be of the same type.")
+
+            if issubclass(first_type, int) and use_int_array:
+                # Check the range of the integers in the list
+                use_int = all(-(1 << 31) <= item < 1 << 31 for item in data)
+                use_long = all(-(1 << 63) <= item < 1 << 63 for item in data)
+                if use_int:
+                    return IntArrayNBT(data, name=name)
+                if not use_long:  # Too big to fit in a long, won't fit in a List of Longs either
+                    raise ValueError("Integer list contains values out of range.")
+                return LongArrayNBT(data, name=name)
+            return ListNBT([NBTag.from_object(item, use_int_array=use_int_array) for item in data], name=name)
+        if isinstance(data, dict):
+            if len(data) == 0:
+                return CompoundNBT([], name=name)
+            if len(data) == 1 and name == "":
+                key, value = next(iter(data.items()))
+                return NBTag.from_object(value, name=key, use_int_array=use_int_array)
+            payload = []
+            for key, value in data.items():
+                tag = NBTag.from_object(value, name=key, use_int_array=use_int_array)
+                payload.append(tag)
+            return CompoundNBT(payload, name)
+        if data is None:
+            warnings.warn("Converting None to an END tag.", stacklevel=2)
+            return EndNBT()  # Should not be used
+
+        try:
+            # Check if the object can be converted to bytes
+            return ByteArrayNBT(bytes(data), name=name)  # type: ignore
+        except (TypeError, ValueError):
+            pass
```

Comment:
`try-except` with empty `except` should be replaced with `contextlib.suppress`:
```suggestion
        with suppress(TypeError, ValueError):
            # Check if the object can be converted to bytes
            return ByteArrayNBT(bytes(data), name=name)  # type: ignore
```